### PR TITLE
[bitnami/milvus] feat: config external kafka tls client certs setting…

### PR DESCRIPTION
**Pull request by chenraoCR**
### Description of the change

config externalKafka tls settings

### Benefits

allow users to config kafka tls client certs when connect external kafka running with tls

### Applicable issues

- feat: https://github.com/bitnami/charts/issues/26110

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/milvus/templates/_helpers.tpl [hunk 1]
```diff
@@ -990,22 +990,41 @@ Init container definition for waiting for the database to be ready
...
+      {{- if not (empty .context.Values.externalKafka.tls.keyPassword) }}
+      yq e -i '.kafka.ssl.tlsKeyPassword = "{{ .context.Values.externalKafka.tls.keyPassword }}"' /bitnami/milvus/rendered-conf/pre-render-config_01.yaml
       {{- end }}
       {{- end }}
       {{- else }}
       mv /bitnami/milvus/rendered-conf/pre-render-config_00.yaml /bitnami/milvus/rendered-conf/pre-render-config_01.yaml
       {{- end }}
-      render-template /bitnami/milvus/rendered-conf/pre-render-config_01.yaml > /bitnami/milvus/rendered-conf/milvus.yaml
+
+      # server tls certs settings
```
Comment (migruiz4) [inline bitnami/milvus/templates/_helpers.tpl:1017]:
```suggestion
      # Milvus server TLS settings
```

File: bitnami/milvus/templates/_helpers.tpl [hunk 2]
```diff
@@ -990,22 +990,41 @@ Init container definition for waiting for the database to be ready
       cp -r /opt/bitnami/milvus/configs/. /bitnami/milvus/rendered-conf
       # Build final milvus.yaml with the sections of the different files
       find /bitnami/milvus/conf -type f -name *.yaml -print0 | sort -z | xargs -0 yq eval-all '. as $item ireduce ({}; . * $item )' /bitnami/milvus/rendered-conf/milvus.yaml > /bitnami/milvus/rendered-conf/pre-render-config_00.yaml
+
+      # kafka settings
```
Comment (migruiz4) [inline bitnami/milvus/templates/_helpers.tpl:994]:
```suggestion
      # Kafka settings
```

File: bitnami/milvus/templates/_helpers.tpl [hunk 3]
```diff
@@ -990,22 +990,41 @@ Init container definition for waiting for the database to be ready
...
       # HACK: In order to enable Kafka we need to remove all Pulsar settings from the configuration file
       # https://github.com/milvus-io/milvus/blob/master/configs/milvus.yaml#L110
       yq 'del(.pulsar)' /bitnami/milvus/rendered-conf/pre-render-config_00.yaml > /bitnami/milvus/rendered-conf/pre-render-config_01.yaml
-      yq e -i '.common.security.tlsMode = {{ .context.Values.proxy.tls.mode }}' /bitnami/milvus/rendered-conf/pre-render-config_01.yaml
-      {{- if ne (int .context.Values.proxy.tls.mode) 0 }}
-      yq e -i '.tls.serverPemPath = "/opt/bitnami/milvus/configs/cert/milvus/{{ .context.Values.proxy.tls.cert }}"' /bitnami/milvus/rendered-conf/pre-render-config_01.yaml
-      yq e -i '.tls.serverKeyPath = "/opt/bitnami/milvus/configs/cert/milvus/{{ .context.Values.proxy.tls.key }}"' /bitnami/milvus/rendered-conf/pre-render-config_01.yaml
-      {{- if eq (int .context.Values.proxy.tls.mode) 2 }}
-      yq e -i '.tls.caPemPath = "/opt/bitnami/milvus/configs/cert/milvus/{{ .context.Values.proxy.tls.caCert }}"' /bitnami/milvus/rendered-conf/pre-render-config_01.yaml
+      # tls settings
```
Comment (migruiz4) [inline bitnami/milvus/templates/_helpers.tpl:999]:
```suggestion
      # Kafka TLS settings
```

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Thank you for your contribution @chenraoCR! I have updated the comments at `milvus.prepareMilvusInitContainer` to use capital letters.

Additionally, I would like to request one change in this PR.

Instead of `{{- if (not (empty .Values.*)) }}` could you please use `{{- if .Values.* }}`? That would simplify the conditionals and help improve code readability.

**Comment by chenraoCR** [comment]
> Thank you for your contribution @chenraoCR! I have updated the comments at `milvus.prepareMilvusInitContainer` to use capital letters.
> 
> Additionally, I would like to request one change in this PR.
> 
> Instead of `{{- if (not (empty .Values.*)) }}` could you please use `{{- if .Values.* }}`? That would simplify the conditionals and help improve code readability.

noted

File: bitnami/milvus/templates/_helpers.tpl (hunk 1 above)
Comment (chenraoCR) [reply bitnami/milvus/templates/_helpers.tpl]: resolved

**Comment by chenraoCR** [comment]
@migruiz4  i committed the PR for quite some time and updated the requested changes, could u please merge it?

**Comment by migruiz4** [comment]
Hi @chenraoCR,

I'm sorry for the inconvenience. We were having an issue with the CI/CD pipeline for Milvus and the tests were not passing due to a problem unrelated to this PR. 

Because the CI/CD pipeline would not succeed we couldn't merge this PR.

**Comment by chenraoCR** [comment]
@migruiz4 now seems okay, plz check and merge, thanks

**Review by migruiz4 — APPROVED** [review]

**Pull request merged**